app/main/views/root.py

```diff
@@ -100,10 +96,11 @@ def _load_and_parse_schema(eq_id, form_type):
     :return: an object model
     """
     # load the schema
-    json_schema = load_schema(eq_id, form_type)
+
+    json_schema = load_schema(current_user.get_eq_id(), current_user.get_form_type())
     if json_schema:
         parser = SchemaParserFactory.create_parser(json_schema)
         schema = parser.parse()
         return schema
     else:
-        return None
+        return errors.page_not_found()
```

Comment:
I don't like this, it should either return a constructed schema model or none. It shouldn't return a rendered page.